docs/blog/_posts/2019-04-10-14th-dotty-milestone-release.md

```diff
@@ -0,0 +1,200 @@
+---
+layout: blog-page
+title: Announcing Dotty 0.14.0-RC1 with export, immutable arrays, creator applications and more
+author: Anatolii Kmetiuk
+authorImg: /images/anatolii.png
+date: 2019-04-10
+---
+
+Hello! This is the 14th release of Dotty. Some of the most interesting changes in this release include the new `export`, the dual of `import`, feature, an immutable array type and the creator applications syntax.
+
+This release serves as a technology preview that demonstrates new
+language features and the compiler supporting them.
+
+Dotty is the project name for technologies that are being considered for
+inclusion in Scala 3. Scala has pioneered the fusion of object-oriented and
+functional programming in a typed setting. Scala 3 will be a big step towards
+realising the full potential of these ideas. Its main objectives are to
+
+- become more opinionated by promoting programming idioms we found to work well,
+- simplify where possible,
+- eliminate inconsistencies and surprising behaviours,
+- build on strong foundations to ensure the design hangs together well,
+- consolidate language constructs to improve the language’s consistency, safety, ergonomics, and
+  performance.
+
+You can learn more about Dotty on our [website](https://dotty.epfl.ch).
+
+<!--more-->
+
+This is our 14th scheduled release according to our
+[6-week release schedule](https://dotty.epfl.ch/docs/usage/version-numbers.html).
+
+# What’s new in the 0.14.0-RC1 technology preview?
+
+## Export as a dual of Import
+
+A new `export` keyword is added to the language that defines aliases for selected members of an object. Consider the following example:
+
+```scala
+class BitMap
+class InkJet
+class Printer {
+  type PrinterType
+  def print(bits: BitMap): Unit = ???
+  def status: List[String] = ???
+}
+class Scanner {
+  def scan(): BitMap = ???
+  def status: List[String] = ???
+}
+class Copier {
+  private val printUnit = new Printer { type PrinterType = InkJet }
+  private val scanUnit = new Scanner
+  export scanUnit.scan
+  export printUnit.{status => _, _}
+  def status: List[String] = printUnit.status ++ scanUnit.status
+}
+```
+
+When defined like this, the `Copier` class defines aliases of the `scanner`'s `scan` method and all the methods of `printUnit` except the `status` method. You can hence call them on the `Copier` as follows:
+
+```scala
+val copier = new Copier
+copier.print(copier.scan())
+```
+
+The motivation for this change is to promote aggregation over inheritance. In OOP languages it is easy to define inheritance but the above example would be tricky if you follow the aggregation route. One would need to implement proxy methods in the `Copier` to delegate to those of the `Scanner` and the `Printer`.

```

Comment:
I think it is more common to refer to this as "composition" instead of "aggregation"? https://en.wikipedia.org/wiki/Composition_over_inheritance

Reply:
In the [docs](https://github.com/lampepfl/dotty/blob/master/docs/docs/reference/other-new-features/export.md) the "aggregation" term is used though. Should we also update it to "composition" there?